### PR TITLE
Add linting github action

### PR DESCRIPTION
**Pull request by tpantelis**

### CODE REVIEW

**Comment by tpantelis** [comment]
hmm... the linting workflow didn't run. @dfarrell07 is this a chicken-and-the-egg thing, ie the file must be added first then will work on subsequent PRs, or did I miss something? What's odd is that the job added by https://github.com/submariner-io/cloud-prepare/pull/7 ran.

**Comment by dfarrell07** [comment]
It should run in this PR, it shouldn't be a chicken-an-egg situation. I wonder if there's a subtle error in the GHA definitions? That causes them to silently not run, as we're seeing here. I could also imagine it just needing a force push or something for GH to pick up the new jobs and run them.

**Comment by tpantelis** [comment]
> It should run in this PR, it shouldn't be a chicken-an-egg situation. I wonder if there's a subtle error in the GHA definitions? That causes them to silently not run, as we're seeing here. I could also imagine it just needing a force push or something for GH to pick up the new jobs and run them.

I'll try that.  I copied this from submariner as is (only removed the code-gen job) so it should be fine.

Tried it - still doesn't run. I'll try paring it down to the simplest job. The issue may be one specific  job.

**Comment by dfarrell07** [comment]
Well, I'm not sure what you did but I see it's working now. :heavy_check_mark:

**Review by dfarrell07 — APPROVED** [review]

File: pkg/aws/gw-machineset.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ kind: MachineSet
 metadata:
   labels:
     machine.openshift.io/cluster-api-cluster: {{.InfraID}}
-  name: {{.InfraID}}-submariner-gw-{{.AZ}}
```
Comment (mkolesnik) [inline pkg/aws/gw-machineset.yaml]: This commented out on purpose?

**Review by mkolesnik — COMMENTED** [review]
Thanks Tom, one question :)

File: pkg/aws/gw-machineset.yaml (hunk 1 above)
Comment (tpantelis) [reply pkg/aws/gw-machineset.yaml]: crap - I didn't mean that - thanks for catching it.  The yaml linter was failing on that line due to the "{{*}}" part so I was playing around with it.  I think it's a false positive and I ended up taking out that linter.

**Review by mangelajo — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**